sdk/servicebus/Microsoft.Azure.ServiceBus/tests/SenderReceiverTests.cs

```diff
@@ -59,7 +59,14 @@ public async Task ReceiveLotOfMessagesWithoutSettling()
 
                 try
                 {
-                    await this.ReceiveDeleteTestCase(sender, receiver, messageCount);
+                    var receiveTasks = new List<Task>();
+                    int batchSize = 10;
+                    // There is no guarantee that we can receive 5010 messages in one batch, so splitting them to different batches.
+                    for (var i = 0; i < messageCount / batchSize; i++)
+                    {
+                        receiveTasks.Add(this.ReceiveDeleteTestCase(sender, receiver, batchSize));
+                    }
+                    await Task.WhenAll(receiveTasks);
```

Comment:
This is not the right change.
I think the test is flakey becuase the test is not receiving all 5010 messages within 10 seconds. I think increasing that timeout to 1 minute for this test alone should fix this.
Method ReceiveDeleteTestCase uses 10 seconds of time. Instead pass timeout to that method with default being 10 seconds. This flakey will pass a timeout of 1 minute or 2 minutes that should fix it.

Reply:
Thanks for the comments, I thought about increasing the timeout but think 5010 messages are still hard to receive in one batch even if we increase the timeout to a larger period if network latency varies in different tests. The goal of this test is to verify that we indeed received these 5010 messages, so breaking them into smaller batches in my opinion can be more reliable in the nightly stress run?

Reply:
It looks like ReceiveDeleteTestCase already handles looping - I didn't realize this when looking at this test initially.

Reply:
Increasing timeout change looks good. 2 minutes is plenty of time for 5000 messages. Receive part already handles multiple calls.